src/api/java/CMakeLists.txt

```diff
@@ -9,8 +9,55 @@
 ## directory for licensing information.
 ##
 
-# create a directory for the cvc package
-file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/cvc")
+# create directories
+file(MAKE_DIRECTORY
+  "${CMAKE_CURRENT_BINARY_DIR}/src/main/resources"
+  "${CMAKE_CURRENT_BINARY_DIR}/src/main/java"
+  "${CMAKE_CURRENT_BINARY_DIR}/src/test/java"
+  )
+
+# find java
+find_package(Java)
+# find jni package
+find_package(JNI REQUIRED)
+
+# specify the output destination of cvcjni library
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/src/main/resources")
+
+add_library(cvcjni SHARED jni/cvc_Solver.cpp)
+
+target_include_directories(cvcjni PUBLIC ${JNI_INCLUDE_DIRS})
+target_include_directories(cvcjni PUBLIC ${PROJECT_SOURCE_DIR}/src)
+target_include_directories(cvcjni PUBLIC ${CMAKE_BINARY_DIR}/src/)
+target_link_libraries(cvcjni ${JNI_LIBRARIES})
+target_link_libraries(cvcjni cvc4)
+
+
+# copy libraries and java files
+add_custom_target(copy-java-api-files
```

Comment:
Why are we copying these files?
Is it so that gradle finds them? Can't we make gradle tell where they are?

(I'd rather like to change gradle so that it plays nicely with cmake instead of modifying cmake to make gradle work in its default config)

Reply:
Cmake is used now for compiling java files. So copying is no longer needed.